### PR TITLE
[DCJ-346][risk=no] Updated linter to support typescript linting

### PR DESCRIPTION
**Pull request by sjkobori**
### Addresses
https://broadworkbench.atlassian.net/browse/DCJ-346
### Summary
Added support for Typescript linting. These changes have been extracted from: https://github.com/DataBiosphere/duos-ui/pull/2557

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -55,10 +55,32 @@
...
+      },
+      "extends": [
+        "eslint:recommended", 
+        "plugin:react/recommended", 
+        "plugin:@typescript-eslint/eslint-recommended"
+      ]
     }
   ]
-}
+}
```
Comment (rushtong) [inline .eslintrc]: nit: revert the removal of the newline

**Review by rushtong — APPROVED** [review]
This looks good, thanks 👍🏽  One minor nit inline.

File: .eslintrc [hunk 2]
```diff
@@ -55,10 +55,32 @@
...
+        "@typescript-eslint/no-unused-vars": [
+          "warn",
+          {
+            "vars": "all",
+            "args": "all",
+            "argsIgnorePattern": "_|^_|^props",
+            "destructuredArrayIgnorePattern": "_|^_",
+            "ignoreRestSiblings": true
+          }
+        ]
```
Comment (fboulnois) [inline .eslintrc]: I think we should only ignore unused variables that start with `_`. Per the eslint docs, the recommended syntax for this is:

```
    "@typescript-eslint/no-unused-vars": [
      "error",
      {
        "args": "all",
        "argsIgnorePattern": "^_",
        "caughtErrors": "all",
        "caughtErrorsIgnorePattern": "^_",
        "destructuredArrayIgnorePattern": "^_",
        "varsIgnorePattern": "^_",
        "ignoreRestSiblings": true
      }
    ]
```

**Review by fboulnois — CHANGES_REQUESTED** [review]
See below:

File: .eslintrc (hunk 2 above)
Comment (sjkobori) [reply .eslintrc]: Sounds good to me. It seems like `caughtErrors` already defaults to `all`, so I won't explicitly add that.

**Review by fboulnois — APPROVED** [review]
lgtm, thank you 👍

**Pull request merged**